src/samples/coap/lwm2m-client.c

```diff
@@ -623,22 +633,47 @@ main(int argc, char *argv[])
         goto exit;
     }
 
+    security_data = calloc(1, sizeof(struct security_obj_instance_ctx));
+    if (!security_data) {
+        fprintf(stderr, "Could not alloc memory for security object context\n");
+        return -ENOMEM;
+    }
+
+    security_data->client = client;
+
     if (!data_ctx.is_bootstrap) {
         r = sol_lwm2m_add_object_instance(client, &server_object, NULL);
+
         if (r < 0) {
             fprintf(stderr, "Could not add a server object instance\n");
             goto exit_del;
         }
+/*
+        security_data->server_uri = (char*)malloc((strlen("coap://localhost:5683") + 1) * sizeof(char));
+        strcpy(security_data->server_uri, "coap://localhost:5683");
+ */
+        security_data->server_uri = sol_str_slice_to_str(sol_str_slice_from_blob(&server_addr));
```

Comment:
`server_addr` is statically defined at the beginning of this file.

Anyway, I went with the changes to make server_uri a `sol blob *` and these things work well and are more clear now -- currently there are no leaks.

I'm currently changing the Server Object of this client to be "dynamic" (instead of hard-coded in the `read_server_obj` function) as well, and to have write and del functions, to retest the second call to `sol_lwm2m_client_start`.